File: SharpForum.API/Models/Domain/Reply.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SharpForum.API.Models.Domain
{
    /// <summary>
    /// Topic reply
    /// </summary>
    public class Reply : Entity
    {
        public Reply()
        {
            Message = string.Empty;
        }

        /// <summary>
        /// Message body
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// Author identifier
        /// </summary>
        public Guid AuthorId { get; set; }

        /// <summary>
        /// Author of the reply
        /// </summary>
        public virtual User Author { get; set; }

        /// <summary>
        /// Related topic identifier
        /// </summary>
        [Required]
        public Guid TopicId { get; set; }

        /// <summary>
        /// Related topic
        /// </summary>
        public virtual Topic Topic { get; set; }
    }
}

File: sharpforum-web/src/app/models/reply.ts
import { Topic } from "./topic";
import { User } from "./user";

export interface Reply {
    id: string;
    message: string;
    authorId: string;
    author: User;
    topicId: string;
    topic: Topic;
    createdOn: Date;
    updatedOn: Date;
}

export interface AddReplyFormValues {
    authorId: string;
    topicId: string;
    message: string;
}